src/services/priceService.js

```
@@ -14,32 +16,65 @@ const {
   processFloatString,
   getTimestamp,
 } = require("../utils/helpers");
+const { logErrorToFile } = require("../utils/errorLogger");
 
 const DEPLOYER_KEY = process.env.DEPLOYER_KEY;
+const signerClient =
+  process.env.MAINNET_SIGNER_CLIENT == undefined
+    ? testnetSignatureClient
+    : process.env.MAINNET_SIGNER_CLIENT == 1
+    ? mainnetSignatureClient
+    : testnetSignatureClient;
 
-async function callSignAPICall(url, resultPath, provider) {
+async function callSignAPICall(
+  url,
+  resultPath,
+  provider,
+  providerWeight,
+  coinId
+) {
   try {
     const config = getHeaderConfig(provider);
-    const response = await axios.get(url, config);
+    const response = await fetchCryptoData(provider, coinId, {
+      useDoh: true,
+      headers: config?.headers || {},
+    });
 
     const price = _.get(response, resultPath);
+    const weightedPrice = price * providerWeight;
```

before that line, lets make sure that the price actually can be used for further computation,  e.g.:
```
if (!price || isNaN(Number(price))) {
  throw new Error(`Invalid price received from ${provider}`);
}
```

---

Do we need a separate error check here for a reason? Asking causes the whole thing in under a try-catch block anyways and since if the price comes corrupted its going to the catch block.
```
 console.error(`Error calling API ${url}:`, error.message);
    await logErrorToFile(
      "API_CALL",
      `Error calling API ${url} for provider ${provider} and coin ${coinId}`,
      error
    );
    return ["0", 0, null, url];
```

---

tbh I'm just not sure if some weird values after being multiplied by the weight will become NaN and silently creep into the rest of the computation. Even if now it is harmless, maybe in the future it will not be. I'd rather fail early and explicitly to reduce the cognitive load on such edge cases.